flink-java/src/main/java/org/apache/flink/api/java/typeutils/runtime/AvroSerializer.java

```diff
@@ -96,27 +93,38 @@ public T createInstance() {
 	@Override
 	public T copy(T from) {
 		checkKryoInitialized();
-		return this.kryo.copy(from);
+
+		try {
+			return kryo.copy(from);
+		} catch (KryoException ke) {
+			// Kryo could not copy the object --> try to serialize/deserialize the object
+			try {
+				byte[] byteArray = InstantiationUtil.serializeToByteArray(this, from);
+
+				return InstantiationUtil.deserializeFromByteArray(this, byteArray);
```

Comment:
Why are you using Java serialization here? Isn't the idea of a specialized Avro serializer to use it?